### PR TITLE
openimageio: default to openimageio2, rename 1.x to openimageio_1

### PR DESCRIPTION
**Pull request by risicle**
###### Description of changes

This moves openimageio 1.x a step towards removal - it is likely vulnerable to many of the CVEs mentioned in #208013

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -31414,11 +31414,11 @@ with pkgs;
 
   openfx = callPackage ../development/libraries/openfx {};
 
-  openimageio = callPackage ../applications/graphics/openimageio {
+  openimageio_1 = callPackage ../development/libraries/openimageio/1.x.nix {
     boost = boost175;
   };
 
-  openimageio2 = darwin.apple_sdk_11_0.callPackage ../applications/graphics/openimageio/2.x.nix { };
+  openimageio = darwin.apple_sdk_11_0.callPackage ../development/libraries/openimageio/default.nix { };
```
Comment (panicgh) [inline pkgs/top-level/all-packages.nix:31421]: Could be shortened:
```suggestion
  openimageio = darwin.apple_sdk_11_0.callPackage ../development/libraries/openimageio { };
```

**Review by panicgh — CHANGES_REQUESTED** [review]
Did you forget to alias `openimageio2` or rename it in the other derivations? These pkgs still use it and break as a result:
* pkgs/applications/misc/blender/default.nix
* pkgs/development/compilers/osl/default.nix

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (risicle) [reply pkgs/top-level/all-packages.nix:31421]: Knew there was something I'd forget...

**Comment by risicle** [comment]
Trying to figure out why it didn't refuse to evaluate then...

**Review by panicgh — APPROVED** [review]
Result of `nixpkgs-review pr 209104` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>luxcorerender</li>
    <li>openimageio_1</li>
  </ul>
</details>
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>blender</li>
    <li>blender-hip</li>
    <li>embree</li>
    <li>embree2</li>
    <li>openimageio</li>
    <li>python310Packages.bpycv</li>
    <li>python39Packages.bpycv</li>
  </ul>
</details>

**Review by LeSuisse — APPROVED** [review]
Changes looks reasonable, `nixpkgs-review` happy on x86_64.

**Pull request merged**